### PR TITLE
Fixes reaction chambers eatting beakers, and lets users eject beakers when out of power with alt.

### PR DESCRIPTION
**Pull request by Thalpy**
## About The Pull Request

Due to an oversight, beakers are eatten on deconstruct, which shouldn't happen. Also allows beakers to be alt clicked out of them to be in line with the dispenser.

## Why It's Good For The Game

Consisency and fixes an unintentional beaker deletion.

## Changelog
:cl:
fix: Fixes reaction chambers eatting beakers, and lets users eject beakers when out of power with alt.
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/machinery/chem_heater.dm [hunk 1]
```diff
@@ -35,13 +35,14 @@
...
+	create_reagents(200, NO_REACT)//Lets save some calculations here
 	//TODO: comsig reaction_start and reaction_end to enable/disable the UI autoupdater - this doesn't work presently as there's a hard divide between instant and processed reactions
-	
+
 /obj/machinery/chem_heater/Destroy()
 	if(beaker)
 		UnregisterSignal(beaker.reagents, COMSIG_REAGENTS_REACTION_STEP)
-		QDEL_NULL(beaker)
+		beaker.forceMove(drop_location())
+		beaker = null
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/machinery/chem_heater.dm:45]: This should be done in deconstruct(). Destroy() should still QDEL_NULL beaker if it's still there

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/machinery/chem_heater.dm (hunk 1 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/machinery/chem_heater.dm:46]: Is this okay?

Comment (Mothblocks) [reply code/modules/reagents/chemistry/machinery/chem_heater.dm:46]: Yes, but bring back the `Destroy` that calls `QDEL_NULL(beaker)`, in the cases of deconstruct not being called.

Comment (Thalpy) [reply code/modules/reagents/chemistry/machinery/chem_heater.dm:46]: Destroy does call QDEL_NULL(beaker)? Am I misunderstanding?

Comment (Cyberboss) [reply code/modules/reagents/chemistry/machinery/chem_heater.dm:46]: If you look at Files Changed it doesn't show that.

Comment (Thalpy) [reply code/modules/reagents/chemistry/machinery/chem_heater.dm:46]: That's how it handles it currently so I didn't change that.

**Review by Cyberboss — APPROVED** [review]

**Comment by Cyberboss** [comment]
My bad, I read deconstruct() as Destroy()

**Pull request merged**